File: PythonFunctions/04_math_utils.py
"""Math utility functions"""
import math


def factorial(n: int) -> int:
    """Calculate factorial"""
    return math.factorial(n)


def gcd(a: int, b: int) -> int:
    """Greatest common divisor"""
    return math.gcd(a, b)


def is_prime(n: int) -> bool:
    """Check if number is prime"""
    if n < 2:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

File: JSFunctions/04_math_utils.js
/**
 * Math utility functions
 */

/**
 * Generate random number in range
 * @param {number} min - Minimum value
 * @param {number} max - Maximum value
 * @returns {number} Random number
 */
function randomInRange(min, max) {
    return Math.floor(Math.random() * (max - min + 1)) + min;
}

/**
 * Check if number is prime
 * @param {number} num - Number to check
 * @returns {boolean} True if prime
 */
function isPrime(num) {
    if (num <= 1) return false;
    for (let i = 2; i <= Math.sqrt(num); i++) {
        if (num % i === 0) return false;
    }
    return true;
}

/**
 * Calculate factorial
 * @param {number} n - Number
 * @returns {number} Factorial
 */
function factorial(n) {
    if (n <= 1) return 1;
    return n * factorial(n - 1);
}

module.exports = { randomInRange, isPrime, factorial };
